File: code/Java/第1章 数据结构与算法/1.3 基础算法/MontePI.java
public class MontePI {
	public static void main(String[] args) {
		System.out.println("PI = " + montePI(10000000));
	}
	
	/**
	 * @brief  蒙特卡洛算法计算PI
	 * @param  n: 样本数量
	 * @retval PI的近似值
	 */
	public static double montePI(int n) {
		int cnt = 0;
		for(int i = 0; i < n; i++) {
	        double x = Math.random();		// [0, 1)
	        double y = Math.random();		// [0, 1)
	        if(Math.sqrt(x*x + y*y) <= 1) {
	            cnt++;
	        }
	    }
	    return 4.0 * cnt / n;
	}
}

File: code/C/Chapter1/1-3/monte_pi.c
#include <stdio.h>
#include <stdlib.h>
#include <time.h>
#include <math.h>

double monte_pi(int n) {
    int cnt = 0;
    for(int i = 0; i < n; i++) {
        double x = rand() / (RAND_MAX + 1.0);   // [0, 1]
        double y = rand() / (RAND_MAX + 1.0);   // [0, 1]
        if(sqrt(x*x + y*y) <= 1) {
            cnt++;
        }
    }
    return 4.0 * cnt / n;
}

int main() {
    srand(time(NULL));
    printf("PI = %f\n", monte_pi(10000000));
    return 0;
}